basic-application/basic-application-core/src/main/java/fr/openwide/core/basicapp/core/security/service/BasicApplicationPermissionEvaluator.java

```diff
@@ -3,22 +3,37 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.acls.model.Permission;
 
+import fr.openwide.core.basicapp.core.business.referential.model.City;
 import fr.openwide.core.basicapp.core.business.user.model.User;
 import fr.openwide.core.jpa.security.service.AbstractCorePermissionEvaluator;
 import fr.openwide.core.jpa.security.service.ISecurityService;
+import fr.openwide.core.jpa.util.HibernateUtils;
 
 public class BasicApplicationPermissionEvaluator extends AbstractCorePermissionEvaluator<User> {
 
 	@Autowired
 	private ISecurityService securityService;
 
+	@Autowired
+	private IDefaultGenericListItemPermissionEvaluator defaultGenericListItemPermissionEvaluator;
+	
 	public BasicApplicationPermissionEvaluator() {
 	}
 
 	@Override
 	protected boolean hasPermission(User user, Object targetDomainObject, Permission permission) {
-		// Déléguer ici les permissions aux différents PermissionEvaluator pour chaque objet métier
-		// Au besoin, court-circuiter les vérifications pour le super user : if(isSuperUser(user) { return true }
+		if (targetDomainObject != null) {
+			targetDomainObject = HibernateUtils.unwrap(targetDomainObject); // NOSONAR
+		}
+		
+		if (user != null) {
+			user = HibernateUtils.unwrap(user); // NOSONAR
+		}
+		
+		if (targetDomainObject instanceof City) {
```

Comment:
I think there still should be a comment here explaining "call your own permissionEvaluators here".
Plus, this line should be `instanceof GenericListItem` since this is supposed to be the default permission evaluator for all generic list items. Same goes for the cast below.